htlcswitch/link.go

```diff
@@ -1041,12 +1041,16 @@ out:
 
 			// We'll check to see if we should update the fee rate
 			// based on our current set fee rate. We'll cap the new
-			// fee rate to our max fee allocation.
+			// fee rate to our max fee allocation and ensure it's
+			// above our fee floor to prevent a channel failure.
 			commitFee := l.channel.CommitFeeRate()
 			maxFee := l.channel.MaxFeeRate(l.cfg.MaxFeeAllocation)
 			newCommitFee := lnwallet.SatPerKWeight(
 				math.Min(float64(netFee), float64(maxFee)),
 			)
+			if newCommitFee < lnwallet.FeePerKwFloor {
```

Comment:
would a better place for this be inside `MaxFeeRate`? I can see keeping this here as a sanity check, but really `MaxFeeRate` shouldn't propose anything lower than `lnwallet.FeePerKwFloor`

Reply:
would also be easier to test

Reply:
Good point, done.